File: test-suite-kotlin-server-generator/src/main/java/io/micronaut/openapi/test/api/ParametersController.java
package io.micronaut.openapi.test.api;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.ZonedDateTime;
import java.util.stream.Collectors;

import io.micronaut.data.model.Pageable;
import io.micronaut.data.model.Sort;
import io.micronaut.http.annotation.Controller;
import io.micronaut.openapi.test.filter.MyFilter;
import io.micronaut.openapi.test.model.ColorEnum;
import io.micronaut.openapi.test.model.SendDatesResponse;
import io.micronaut.openapi.test.model.SendPrimitivesResponse;

import org.jetbrains.annotations.NotNull;

import reactor.core.publisher.Mono;

@Controller
public class ParametersController implements ParametersApi {

    @Override
    public Mono<SendPrimitivesResponse> sendPrimitives(String name, BigDecimal age, float height, boolean isPositive) {
        var response = new SendPrimitivesResponse();
        response.setName(name);
        response.setAge(age);
        response.setHeight(height);
        response.setPositive(isPositive);
        return Mono.just(response);
    }

    @NotNull
    @Override
    public Mono<String> sendValidatedPrimitives(String name, Integer age, BigDecimal favoriteNumber, Double height) {
        return Mono.just("Success");
    }

    @Override
    public Mono<SendDatesResponse> sendDates(LocalDate commitDate, ZonedDateTime commitDateTime) {
        var response = new SendDatesResponse();
        response.setCommitDate(commitDate);
        response.setCommitDateTime(commitDateTime);
        return Mono.just(response);
    }

    @Override
    public Mono<ColorEnum> sendParameterEnum(ColorEnum colorParam) {
        return Mono.just(colorParam);
    }

    @Override
    public Mono<String> getIgnoredHeader() {
        return Mono.just("Success");
    }

    @Override
    public Mono<String> sendIgnoredHeader() {
        return Mono.just("Success");
    }

    @Override
    public Mono<String> sendPageQuery(Pageable pageable) {
        return Mono.just(
            "(page: " + pageable.getNumber() +
                ", size: " + pageable.getSize() +
                ", sort: " + sortToString(pageable.getSort()) + ")"
        );
    }

    @Override
    public Mono<String> sendMappedParameter(MyFilter myFilter) {
        return Mono.just(myFilter.toString());
    }

    private String sortToString(Sort sort) {
        return sort.getOrderBy().stream()
            .map(
                order -> order.getProperty() + "(dir=" + order.getDirection() + ")"
            ).collect(Collectors.joining(" "));
    }
}

File: test-suite-kotlin-kapt-server-generator/src/main/kotlin/io/micronaut/openapi/test/api/ParametersController.kt
package io.micronaut.openapi.test.api

import io.micronaut.data.model.Pageable
import io.micronaut.data.model.Sort
import io.micronaut.http.annotation.Controller
import io.micronaut.openapi.test.filter.MyFilter
import io.micronaut.openapi.test.model.ColorEnum
import io.micronaut.openapi.test.model.SendDatesResponse
import io.micronaut.openapi.test.model.SendPrimitivesResponse
import reactor.core.publisher.Mono
import java.math.BigDecimal
import java.time.LocalDate
import java.time.ZonedDateTime

@Controller
open class ParametersController : ParametersApi {

    override fun sendPrimitives(name: String, age: BigDecimal, height: Float, isPositive: Boolean): Mono<SendPrimitivesResponse> {
        return Mono.just(SendPrimitivesResponse(name, age, height, isPositive))
    }

    override fun sendValidatedPrimitives(name: String?, age: Int?, favoriteNumber: BigDecimal?, height: Double?): Mono<String> {
        return Mono.just("Success")
    }

    override fun sendDates(commitDate: LocalDate?, commitDateTime: ZonedDateTime?): Mono<SendDatesResponse> {
        return Mono.just(SendDatesResponse(commitDate, commitDateTime))
    }

    override fun sendParameterEnum(colorParam: ColorEnum?): Mono<ColorEnum> {
        return Mono.just(colorParam!!)
    }

    override fun getIgnoredHeader(): Mono<String> {
        return Mono.just("Success")
    }

    override fun sendIgnoredHeader(): Mono<String> {
        return Mono.just("Success")
    }

    override fun sendPageQuery(pageable: Pageable): Mono<String> {
        return Mono.just(
                "(page: " + pageable.number +
                        ", size: " + pageable.size +
                        ", sort: " + sortToString(pageable.sort) + ")"
        )
    }

    override fun sendMappedParameter(myFilter: MyFilter): Mono<String> {
        return Mono.just(myFilter.toString())
    }

    private fun sortToString(sort: Sort): String {
        return sort.orderBy.joinToString(" ") { order -> order.property + "(dir=" + order.direction + ")" }
    }
}
